core-bundle/src/Migration/CommandCompiler.php

```diff
@@ -28,13 +28,45 @@ public function __construct(private readonly Connection $connection, private rea
     /**
      * @return list<string>
      */
-    public function compileCommands(): array
+    public function compileCommands(bool $doNotDropColumns = false): array
     {
         // Get a list of SQL commands from the schema diff
         $schemaManager = $this->connection->createSchemaManager();
         $fromSchema = $schemaManager->createSchema();
         $toSchema = $this->schemaProvider->createSchema();
 
+        // If columns should not get dropped, we copy missing definitions
+        // over to the $toSchema, so that they won't appear in the diff
+        if ($doNotDropColumns) {
+            foreach ($fromSchema->getTables() as $table) {
+                $toSchemaTable = $toSchema->getTable($table->getName());
```

Comment:
Do we need to check hasTable here?

Reply:
see https://github.com/contao/contao/pull/4888#discussion_r908320116